containers/registry-mock-server/openapi/src/index.ts

```diff
@@ -0,0 +1,215 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as express from 'express';
+import * as cors from 'cors';
+import {RegistryClient} from '@giteshk-org/apigeeregistry';
+import {ClientOptions} from 'google-gax';
+import * as grpc from '@grpc/grpc-js';
+const Yaml = require('js-yaml');
+
+import {IHttpOperation} from '@stoplight/types';
+import {Request, Response} from 'express';
+
+const {
+  getHttpOperationsFromSpec,
+} = require('@stoplight/prism-cli/dist/operations');
+const {
+  createClientFromOperations,
+} = require('@stoplight/prism-http/dist/client');
+
+const app = express();
+const PORT = process.env.PORT || 3000;
+const HEADER_REGISTRY_SPEC =
+  process.env.HEADER_REGISTRY_SPEC || 'apigee-registry-spec';
+
+const client_options = <ClientOptions>{};
+
+if (process.env.APG_REGISTRY_INSECURE) {
+  if (process.env.APG_REGISTRY_INSECURE === '1') {
+    client_options.sslCreds = grpc.credentials.createInsecure();
+  }
+}
+
+if (process.env.APG_REGISTRY_ADDRESS) {
+  const items = process.env.APG_REGISTRY_ADDRESS.split(':');
+  client_options.apiEndpoint = items[0];
+  client_options.port = items.length >= 1 ? parseInt(items[1]) : 443;
+}
+
+const client = new RegistryClient(client_options);
+
+app.use(cors());
+
+/**
+ * Error Handling
+ *
+ * @param err
+ * @param res
+ */
+function _sendError(err: Error, res: Response) {
+  console.error('Error ' + err.message);
+  res.status(500).send(new Error(err.message)).end();
+}
+
+/**
+ * Process the mock request
+ * @param req
+ * @param res
+ */
+function processMockRequest(req: Request, res: Response) {
```

Comment:
Nit: It might be easier to understand how this method behaves if there is an example in the comments. Essentially to understand how one server is creating mocks for different APIs.

Reply:
I added more details in the comments.